src/kcmd/grub.go

```diff
@@ -23,97 +20,19 @@ func UpdateGrub(cfg *model.InternalConfig) ([]string, error) {
-		return nil, fmt.Errorf("invalid existing parameters in %s: %v", cfg.GrubCfg.GrubDefaultFilePath, err)
-	}
-
-	// Merge parameters while preserving order (existing first, then new in original order)
-	cfg.GrubCfg.Cmdline = model.MergeWithOrderPreservation(existingCmdline, cfg.Data.KernelCmdline)
+	cfg.GrubCfg.Cmdline = strings.Join(cfg.Data.KernelCmdline.Parameters, " ")
 
 	if err := processFile(cfg.GrubCfg); err != nil {
 		return nil, fmt.Errorf("error updating %s: %v", cfg.GrubCfg.CustomGrubFilePath, err)
 	}
 
-	return GrubConclusion(cfg.GrubCfg.CustomGrubFilePath, existingCmdlineStr, cfg.GrubCfg.Cmdline), nil
-}
-
-// parseGrubCMDLineLinuxDefault extracts GRUB_CMDLINE_LINUX_DEFAULT from the GRUB config file.
-func parseGrubCMDLineLinuxDefault(path string) (string, error) {
-	grubMap, err := ParseDefaultGrubFile(path)
-	if err != nil {
-		return "", fmt.Errorf("failed to parse grub file: %v", err)
-	}
-
-	cmdline, ok := grubMap["GRUB_CMDLINE_LINUX_DEFAULT"]
-	if !ok {
-		log.Printf("GRUB_CMDLINE_LINUX_DEFAULT not found in %s", path)
-		return "", nil // Return empty string, not an error
-	}
-
-	return cmdline, nil
-}
-
-// ParseDefaultGrubFile parses a GRUB default configuration file into key-value pairs.
-func ParseDefaultGrubFile(f string) (params map[string]string, err error) {
-	file, err := os.Open(f)
-	if err != nil {
-		return nil, fmt.Errorf("failed to open file %s: %v", f, err)
-	}
-	defer file.Close()
-
-	params = make(map[string]string)
-	scanner := bufio.NewScanner(file)
-	lineNum := 0
-
-	for scanner.Scan() {
-		lineNum++
-		line := strings.TrimSpace(scanner.Text())
-
-		// Skip empty lines and comments
-		if line == "" || strings.HasPrefix(line, "#") {
-			continue
-		}
-
-		// Split the line into key and value
-		parts := strings.SplitN(line, "=", 2)
-		if len(parts) != 2 {
-			log.Printf("Warning: skipping invalid line %d in %s: %s", lineNum, f, line)
-			continue
-		}
-
-		// Trim spaces and quotes from the key and value
-		key := strings.TrimSpace(parts[0])
-		value := strings.TrimSpace(parts[1])
-		value = utils.TrimSurroundingQuotes(value)
-
-		// Warn about duplicates
-		if existing, exists := params[key]; exists {
-			log.Printf("Warning: duplicate key %s in %s (line %d). Previous value: %s, new value: %s",
-				key, f, lineNum, existing, value)
-		}
-
-		params[key] = value
-	}
-
-	if err := scanner.Err(); err != nil {
-		return nil, fmt.Errorf("error reading grub file %s: %v", f, err)
-	}
-
-	return params, err
+	return GrubConclusion(cfg.GrubCfg.CustomGrubFilePath, cfg.GrubCfg.Cmdline), nil
 }
 
 // processFile writes the GRUB configuration to the specified file.
 var processFile = func(grub model.Grub) error {
 	banner := "# This file is automatically generated by rt-conf, please do not edit\n"
-	cmdline := fmt.Sprintf("GRUB_CMDLINE_LINUX_DEFAULT=\"%s\"\n", grub.Cmdline)
+	cmdline := fmt.Sprintf("GRUB_CMDLINE_LINUX_DEFAULT=\"${GRUB_CMDLINE_LINUX_DEFAULT} %s\"\n", grub.Cmdline)
```

Comment:
A little more readable:
```suggestion
	cmdline := fmt.Sprintf(`GRUB_CMDLINE_LINUX_DEFAULT="${GRUB_CMDLINE_LINUX_DEFAULT} %s"\n`, grub.Cmdline)
```